gs_chunked_io/writer.py

```diff
@@ -35,6 +41,13 @@ def __init__(self,
         self._current_part_number = 0
         self._closed = False
         self._upload_id = uuid.uuid4()
+        if threads is not None:
+            assert 2 <= threads
+            max_workers = max(threads, 8)
```

Comment:
Why isn't `max_workers == threads`?

Reply:
At the end of the write process many compose and delete API calls are made to Google Storage. I've found it useful to use 8 threads for these calls, which don't transfer large amounts of data.

The actual part uploads still only use `threads` threads.